### PR TITLE
Adds all rollup REST APIs, updates mappings, adds test resources, ref…

### PR DESCRIPTION
**Pull request by dbbaughe**
…actors generic utils

*Issue #, if available:*

*Description of changes:*
Adds all rollup REST APIs
Updates mappings
Adds test resources
Refactors generic utils

There are some tests that were removed as the runner logic doesn't exist yet, will add those back after

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/explain/ExplainRollupResponse.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ *  * A copy of the License is located at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * or in the "license" file accompanying this file. This file is distributed
+ *  * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  * express or implied. See the License for the specific language governing
+ *  * permissions and limitations under the License.
+ *
+ */
```
Comment (thalurur) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/explain/ExplainRollupResponse.kt:16]: minor formatting

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/start/TransportStartRollupAction.kt [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    private val log = LogManager.getLogger(javaClass)
+
+    override fun doExecute(task: Task, request: StartRollupRequest, actionListener: ActionListener<AcknowledgedResponse>) {
+        val getReq = GetRollupRequest(request.id(), null)
+        client.execute(GetRollupAction.INSTANCE, getReq, object : ActionListener<GetRollupResponse> {
+            override fun onResponse(response: GetRollupResponse) {
+                val rollup = response.rollup
+                if (rollup == null) {
+                    return actionListener.onFailure(
+                        ElasticsearchStatusException("Could not find find rollup [${request.id()}]", RestStatus.NOT_FOUND)
```
Comment (thalurur) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/start/TransportStartRollupAction.kt:58]: minor: reword "Could not find rollup"

File: src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/resthandler/RestExplainRollupAction.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ *  * A copy of the License is located at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * or in the "license" file accompanying this file. This file is distributed
+ *  * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  * express or implied. See the License for the specific language governing
+ *  * permissions and limitations under the License.
+ *
+ */
```
Comment (thalurur) [inline src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/resthandler/RestExplainRollupAction.kt:16]: minor: formatting

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/RollupRestTestCase.kt [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+ *  * A copy of the License is located at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * or in the "license" file accompanying this file. This file is distributed
+ *  * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  * express or implied. See the License for the specific language governing
+ *  * permissions and limitations under the License.
+ *
+ */
```
Comment (thalurur) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/RollupRestTestCase.kt:16]: minor: formatting

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/ActionTests.kt [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ *  * A copy of the License is located at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * or in the "license" file accompanying this file. This file is distributed
+ *  * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  * express or implied. See the License for the specific language governing
+ *  * permissions and limitations under the License.
+ *
+ */
```
Comment (thalurur) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/ActionTests.kt:16]: minor: formatting

File: src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/RequestTests.kt [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+ *  * A copy of the License is located at
+ *  *
+ *  * http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * or in the "license" file accompanying this file. This file is distributed
+ *  * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  * express or implied. See the License for the specific language governing
+ *  * permissions and limitations under the License.
+ *
+ */
```
Comment (thalurur) [inline src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/action/RequestTests.kt:16]: minor: formatting

**Review by thalurur — APPROVED** [review]

**Review by qreshi — APPROVED** [review]

**Pull request merged**